File: Applied Course/4.Problem Solving/3.Problems on Linked List/26.Reverse Linked List.cpp
#include<bits/stdc++.h>
using namespace std;
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *prev=NULL;
        ListNode *current=head;
        ListNode *next=NULL;
        while(current!=NULL)
        {
            next=current->next;
            current->next=prev;
            prev=current;
            current=next;
        }
        return prev;
    }
};
int main()
{
    // https://leetcode.com/problems/reverse-linked-list/
    return 0;
}

File: Applied Course/4.Problem Solving/3.Problems on Linked List/26.Reverse Linked List.py
# https://leetcode.com/problems/reverse-linked-list/
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev=None
        current=head
        next=None
        while current!=None:
            next=current.next
            current.next=prev
            prev=current
            current=next
        return prev